lib/library.cpp

```diff
@@ -1064,21 +1064,27 @@ bool Library::isuninitargbad(const Token *ftok, int argnr, int indirect, bool *h
 /** get allocation info for function */
 const Library::AllocFunc* Library::getAllocFuncInfo(const Token *tok) const
 {
-    const std::string funcname = getFunctionName(tok);
+    std::string funcname = getFunctionName(tok);
+    if (tok->isCpp() && startsWith(funcname, "std::"))
```

Comment:
should we really do this.. what if there is a function in the global scope with the same name with other behavior.

Reply:
This affects only calls to allocation functions like `std::free` etc. which also exist in the global namespace. I don't think there are any collisions.
The `<memory>` block can't handle names like `free,std::free` as is supported elsewhere in the library.